server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1717,7 +1717,6 @@ public void revokeEntitlements(List<Entitlement> entsToRevoke, Set<String> alrea
             int entQuantity = ent.getQuantity() != null ? ent.getQuantity() : 0;
 
             pool.setConsumed(pool.getConsumed() - entQuantity);
-            pool.getEntitlements().remove(ent);
```

Comment:
If we're going to drop the runtime consistency from this operation anyway, it might be faster to just craft three update queries for removing the entitlements and update the pool/consumer consumed/exported values.